Mhv 66332 progress tracker detail page

robertroddev opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No

## Summary

This PR makes it so the process list component will render different content based on different param criteria that is passed to it via props. 

this is hidden behind the `mhv_medications_display_refill_progress` feature flag 

## Related issue(s)

[Add progress tracker of the refill within the details page](https://jira.devops.va.gov/browse/MHV-66332)

## Testing done
unit test

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

| conditions for step |  step displayed |
| ------- | ----- |
| dispStatus = Active: Submitted  |![image](https://github.com/user-attachments/assets/13e83a5b-b254-4e61-9c0a-6395e7e7eec6)|
| dispStatus = 'Active: Refill in Process' |<img width="1233" alt="image" src="https://github.com/user-attachments/assets/19c04f0f-fd93-48f1-948f-3e90e4ac7921" />|
| dispStatus = 'Active' |![image](https://github.com/user-attachments/assets/f8823436-89e4-4397-bba0-7409314aab24)|
| dispStatus = 'Active' and prescription.trackingList[0].completeDateTime is present |![image](https://github.com/user-attachments/assets/9c57506d-7180-4f97-acea-6a6c8c7a1e14)|
| dispStatus = 'Active: Submitted' and prescription.trackingList[0] is present |![image](https://github.com/user-attachments/assets/dd086652-9406-4fa2-a7f8-97761afd3066)|

## What areas of the site does it impact?

mhv-medications

## Acceptance criteria

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

robertroddev committed:
[MHV 66332] modify process list to render different content based on params, while keeping the static list content as well.

robertroddev committed:
[MHV 66332] add process list to VaPrescriptions.

robertroddev committed:
[MHV 66332] extracted tracking configuration to constants file.

robertroddev committed:
[MHV 66332] modified dateFormat helper function to account for time format given by the trackingList object.

robertroddev committed:
Merge branch 'main' into mhv-66332-progress-tracker-detail-page

robertroddev committed:
[MHV 66332] updated unit test for process list.

Commit:
e2e progress tracker

added test to validate progress tracker for submitted, rx in process and shipped

Commit:
e2e updates for tracking

fixed the feature toggles for tracking tests

robertroddev committed:
[MHV 66332] fixed issue with double heading, and missing alert.

Commit:
e2e tracker and delay alert update

Commit:
e2e tracking updates

fixed tracking dispensed date

robertroddev committed:
Merge branch 'main' into mhv-66332-progress-tracker-detail-page

robertroddev committed:
[MHV 66332] updated unit test with proper spelling of Active: Refill in Process.

vmenshutin-bylight commented on src/applications/mhv-medications/components/PrescriptionDetails/VaPrescription.jsx:
```diff
@@ -83,6 +102,56 @@ const VaPrescription = prescription => {
     }
   };
 
+  const stepGuideProps = {
```
can this be moved into `ProcessList`?

robertbylight replied:
yes good idea, i just pushed this 👍

vmenshutin-bylight commented on src/applications/mhv-medications/components/PrescriptionDetails/VaPrescription.jsx:
```diff
@@ -55,8 +56,26 @@ const VaPrescription = prescription => {
+    prescription?.dispStatus === 'Active: Submitted';
   const isRefillRunningLate = isRefillTakingLongerThanExpected(prescription);
 
+  const isCriteriaMetToDisplayProcessList = () => {
```
can this be moved into `ProcessList`?

robertbylight replied:
yes another good idea, i just pushed this 👍

robertroddev committed:
[MHV 66332] removed comments.

rmessina1010 reviewed (dismissed):
LGTM

robertroddev committed:
[MHV 66332] cleaned up VaPrescription by moving logic to ProcessList, fixed unit tests to reflect that refactor. Moved ProcessList under delay alert.

Commit:
e2e element locator updates

fixed element locator and added test to validate rx submitted with no tracking info  and has progress tracker
